app/code/ZP/LoyaltyProgram/Block/Adminhtml/LoyaltyProgram/Edit/AbstractButton.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare(strict_types=1);
+
+namespace ZP\LoyaltyProgram\Block\Adminhtml\LoyaltyProgram\Edit;
+
+use Magento\Backend\Block\Widget\Context;
+use Magento\Framework\View\Element\UiComponent\Control\ButtonProviderInterface;
+use ZP\LoyaltyProgram\Setup\Patch\Data\AddBasicPrograms as BasicProgramsConfig;
+use ZP\LoyaltyProgram\Api\Data\LoyaltyProgramInterface;
+use Magento\Framework\App\RequestInterface;
+
+abstract class AbstractButton implements ButtonProviderInterface
+{
+    protected ?int $programId = null;
+    protected string $requestAction;
+    protected RequestInterface $request;
+
+    public const BUTTON_DATA = [];
+
+    public function __construct(protected Context $context)
+    {
+        $this->request = $this->context->getRequest();
+        $this->setProgramIdPropertyValue();
+        $this->setActionPropertyValue();
+    }
+
+
+    protected function validateButtonWorkConditions(): bool
+    {
+        return !$this->isEditAction() || $this->validateProgramId();
+    }
+
+    protected function isEditAction(): bool
+    {
+        return $this->requestAction === 'edit';
```

Comment:
what is the purpose if this protected function here?

Reply:
The purpose of this method, is to check what type of action is in request. Because for example Delete button will not be shown in New Action ( creating of a new program), but it will be shown in Edit Action.

Reply:
why can't you just use $this->request->getActionName() ?